File: jobs/2026-04-30__17-29-19/node_js_streaming_query_executio__za7pTem/artifacts/code/query.js
const { createClient } = require('@clickhouse/client');

async function run() {
  let host = process.env.CLICKHOUSE_HOST || 'https://localhost:8443';
  if (!host.startsWith('http')) {
    host = `https://${host}`;
  }
  if (!host.includes(':') || host.split(':').length < 3) {
    // If port is not specified, add 8443 as per requirements
    host = host.replace(/\/$/, '');
    if (!host.match(/:\d+$/)) {
      host = `${host}:8443`;
    }
  }

  const client = createClient({
    url: host,
    username: process.env.CLICKHOUSE_USER || 'default',
    password: process.env.CLICKHOUSE_PASSWORD || '',
    database: 'system',
  });

  try {
    const resultSet = await client.query({
      query: 'SELECT * FROM system.tables LIMIT 10',
      format: 'JSONEachRow',
    });

    const stream = resultSet.stream();
    const result = [];

    for await (const rows of stream) {
      rows.forEach((row) => {
        result.push(row.json());
      });
    }

    console.log(JSON.stringify(result, null, 2));
  } catch (error) {
    console.error('Error executing query:', error);
    process.exit(1);
  } finally {
    await client.close();
  }
}

run();

File: jobs/2026-04-30__17-15-44/ch_python_cold_start_handler__ferwaaD/artifacts/code/query.py
import os
import clickhouse_connect

def main():
    host = os.environ.get('CH_HOST', 'localhost')
    port = int(os.environ.get('CH_PORT', 8123))
    user = os.environ.get('CH_USER', 'default')
    password = os.environ.get('CH_PASSWORD', '')

    client = clickhouse_connect.get_client(
        host=host,
        port=port,
        username=user,
        password=password,
        connect_timeout=30,
        send_receive_timeout=120
    )

    result = client.query('SELECT 1')
    print(result.result_rows)

if __name__ == '__main__':
    main()
